bot/seasons.py

```diff
@@ -160,3 +158,24 @@ def get_season(name: str) -> t.Optional[t.Type[SeasonBase]]:
 
         if name in matches:
             return season
+
+
+def _validate_season_overlap() -> None:
+    """
+    Raise BrandingError if there are any colliding seasons.
+
+    This serves as a local test to ensure that seasons haven't been misconfigured.
+    """
+    month_to_season = {}
+
+    for season in SeasonBase.__subclasses__():
+        for month in season.months:
+            colliding_season = month_to_season.get(month)
+
+            if colliding_season:
+                raise BrandingError(f"Season {season} collides with {colliding_season} in {month.name}")
```

Comment:
Would be more convenient if it aggregated all conflicts rather than aborting at the first sight of one. I'm fine with this though.